### PR TITLE
Misc clippy fixes

### PR DESCRIPTION
**Pull request by xangelix**
Just some small clippy fixes to deal with almost all the warnings (at the pedantic level).

Feel free to accept only a portion of this if you don't like the `Self` references for example. I think the rest of the changes look good. :)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -69,7 +69,7 @@ pub struct Style {
 impl Default for Style {
     #[inline]
     fn default() -> Self {
-        Style {
+        Self {
```
Comment (zakarumych) [inline src/lib.rs:72]: I don't like using `Self` in this position.

**Review by zakarumych — CHANGES_REQUESTED** [review]
I'm OK with all of it except one instance.

**Comment by xangelix** [comment]
Understandable-- done!

I'll go ahead and revert the `Self` references in the value pos for https://github.com/zakarumych/egui-snarl/pull/47 as well.

**Comment by zakarumych** [comment]
Sorry, I mean in all those places, not just one :)

File: src/map.rs [hunk 2]
```diff
@@ -35,7 +35,7 @@ impl HashMapProbe {
             .clone()
         });
 
-        HashMapProbe {
+        Self {
```
Comment (zakarumych) [inline src/map.rs:38]: Here

File: src/widget.rs [hunk 3]
```diff
@@ -28,7 +28,7 @@ impl ProbeHeader {
 
         let openness = cx.animate_bool(id, state.open);
 
-        ProbeHeader {
+        Self {
```
Comment (zakarumych) [inline src/widget.rs:31]: Here

File: src/widget.rs [hunk 4]
```diff
@@ -98,9 +98,9 @@ pub struct ProbeLayout {
 }
 
 impl ProbeLayout {
-    fn load(cx: &egui::Context, id: egui::Id) -> ProbeLayout {
+    fn load(cx: &egui::Context, id: egui::Id) -> Self {
         let state = cx.data_mut(|d| *d.get_temp_mut_or(id, ProbeLayoutState { labels_width: 0.0 }));
-        ProbeLayout {
+        Self {
```
Comment (zakarumych) [inline src/widget.rs:103]: Here

**Review by zakarumych — CHANGES_REQUESTED** [review]
And add `![allow(clippy:use_self)]` to the top crate attributes so that it won't trigger the warning if someone runs pedantic clippy.

**Comment by xangelix** [comment]
Sorry about that, definitively not intentional! Looks like some more of my changes didn't go through yesterday as well. I reorganized them into a few different commits.

In particular not so sure about the last two:

- Did you want to use `Self` in pattern matching? Technically this is still under the `clippy::use_self` lint. https://github.com/zakarumych/egui-probe/pull/9/commits/970f881d8bb6772dd2c8b98bcd36590760114890 (would be glad to set allow for the entire proc crate if not)
- Do you care about `clippy::similar_names`? Seems like a dumb change, but I'm happy to change the `_col` wording, or just crate-wide allow that lint as well. https://github.com/zakarumych/egui-probe/pull/9/commits/ce7eaa37700962fa9ae197892dec27fba0bfcaa3

**Comment by zakarumych** [comment]
Yes, I don't like `clippy::use_self`, and it has lots of issues so was send to "nursery" group.
So I'm not sure why clippy with "pedantic" even uses it

**Comment by zakarumych** [comment]
As to `fields_col_probe`.
Which name was too similar to `fields_probe`? And what's `col`? Column? Collider? Colon? :)

**Comment by xangelix** [comment]
> Yes, I don't like `clippy::use_self`, and it has lots of issues so was send to "nursery" group. So I'm not sure why clippy with "pedantic" even uses it

Does that mean you want to keep the pattern matching `Self`s or not?

> As to `fields_col_probe`. What's `col`? Column? Collider? Colon? :)

"Collection"! lol
But honestly whatever is fine. I agree that pendantic has tons of great stuff in there compared to some dumb ones that seem to always come up... eg. complex types, similar names, too many args, use_self kinda iffy too

**Comment by zakarumych** [comment]
`clippy::use_self` is not in `pedantic` anymore anyway. So whatever changes it brings - please, undo them.

Maybe something like `all_fields_probe` would be less confusing?

**Comment by xangelix** [comment]
All done.

**Comment by zakarumych** [comment]
Seems all right!

**Pull request merged**